portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,19 +34,30 @@
     
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-     //retrieves comment data from datastore
+     // Get the user input on how many comments to display.
+     // If user input for max comments is greater than all comments, all comments will display.
+     // If comments are empty, no comments will be displayed.
+     
+    Integer maxComments = getNumDisplayComments(request);
+     // Retrieves comment data from datastore. 
+     // Displays most recent comments first.
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    // Create arraylist of string to store comments.
     ArrayList<String> comments = new ArrayList<String>();
```

Comment:
You can usually omit the type here and let the compiler infer it:

```suggestion
    ArrayList<String> comments = new ArrayList<>();
```